### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by Roland-Ntwali**
## In this PR:
**I created a app. rb file that will serve as your console app entry point. with methods that do the following:**

- List all books.

- List all people.

- Create a person (teacher or student, not a plain Person).

- Create a book.

- Create a rental.

- List all rentals for a given person's id.

 **I created main.rb define the entry point, with method that calls main that is invoked at the end of a file. This method does the following:**

- Present the user with a list of options to perform.

- Lets users choose an option.

- If needed, ask for parameters for the option.

- Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  def create_student
+    puts 'create a new student'
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
```
Comment (AmaduKamara) [inline app.rb]: - [ ] When creating a student, the app crashes whenever I enter a value for the Has parent permission. Kindly work on that so a value for the parent permission question is entered and a student is saved successfully. 👍🏽

Screenshot 2023-06-22 at 4 53 56 PM

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  def create_teacher
+    puts 'create a new teacher'
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @persons << teacher
+    puts 'Teacher created successfully'
+  end
```
Comment (AmaduKamara) [inline app.rb]: - [ ] Good job so far, but when entering details for a Teacher, we're expected to enter age, name, and specialization. Currently, your implementation only captures age and specialization and you have left out the name of a Teacher. Kindly add the name of a Teacher. 👍🏽
- [ ] The app crashes as soon as I enter the value for the specialization. Please work on these so a name for a teacher is captured and a teacher is created successfully. 👍🏽

Screenshot 2023-06-22 at 4 56 45 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @Roland-Ntwali, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽
✅ The Linter checks passed successfully 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Currently, I'm unable to test the rentals feature for renting a book for a person because I'm not able to create a person. Kindly work on the changes so all other features work fine as expected 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      @persons << student
+      puts 'Student doesn\'t have parent permission, can\'t rent books'
+    when 'y'
+      student = Student.new(age: age, name: name, parent_permission: parent_permission, classroom: @classroom)
+      @persons << student
+      puts 'Student created successfully'
+    end
+  end
+
+  def create_teacher
```
Comment (victorokoroji) [inline app.rb:82]: - [ ] As suggested by the previous reviewer,  Good job so far, when entering details for a Teacher, we're expected to enter age, name, and specialization. Currently, your implementation only captures age and specialization and you have left out the name of a Teacher. Kindly add the name of a Teacher. 👍🏽

image

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @Roland-Ntwali 

Great work on making the changes requested by a previous reviewer 👏 🔥 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



## Top highlight 🎯

- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 
- Linters are working properly ✅ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by amejid — APPROVED** [review]
Hi @Roland-Ntwali, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- Console app working as expected :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**